src/org/ods/component/HelmDeploymentStrategy.groovy

```diff
@@ -0,0 +1,260 @@
+    HelmDeploymentStrategy(
+        def script,
+        IContext context,
+        Map<String, Object> config,
+        OpenShiftService openShift,
+        JenkinsService jenkins,
+        ILogger logger
+    ) {
+
+        if (!config.selector) {
+            config.selector = context.selector
+        }
+        if (!config.imageTag) {
+            config.imageTag = context.shortGitCommit
+        }
+        if (!config.deployTimeoutMinutes) {
+            config.deployTimeoutMinutes = context.openshiftRolloutTimeout ?: 15
+        }
+        if (!config.deployTimeoutRetries) {
+            config.deployTimeoutRetries = context.openshiftRolloutTimeoutRetries ?: 5
+        }
+        // Helm options
+        if (!config.chartDir) {
+            config.chartDir = 'chart'
+        }
+        if (!config.containsKey('helmReleaseName')) {
+            config.helmReleaseName = context.componentId
+        }
+        if (!config.containsKey('helmValues')) {
+            config.helmValues = [:]
+        }
+        if (!config.containsKey('helmValuesFiles')) {
+            config.helmValuesFiles = ['values.yaml']
+        }
+        if (!config.containsKey('helmEnvBasedValuesFiles')) {
+            config.helmEnvBasedValuesFiles = []
+        }
+        if (!config.containsKey('helmDefaultFlags')) {
+            config.helmDefaultFlags = ['--install', '--atomic']
+        }
+        if (!config.containsKey('helmAdditionalFlags')) {
+            config.helmAdditionalFlags = []
+        }
+        if (!config.containsKey('helmDiff')) {
+            config.helmDiff = true
+        }
+        if (!config.helmPrivateKeyCredentialsId) {
+            config.helmPrivateKeyCredentialsId = "${context.cdProject}-helm-private-key"
+        }
+        this.script = script
+        this.context = context
+        this.logger = logger
+        this.steps = new PipelineSteps(script)
+
+        this.options = new RolloutOpenShiftDeploymentOptions(config)
+        this.openShift = openShift
+        this.jenkins = jenkins
+    }
+
+    @Override
+    Map<String, List<PodData>> deploy() {
+        if (!context.environment) {
+            logger.warn 'Skipping because of empty (target) environment ...'
+            return [:]
+        }
+
+        def deploymentResources = openShift.getResourcesForComponent(
+            context.targetProject, DEPLOYMENT_KINDS, options.selector
+        )
+        def refreshResources = false
+
+        // Tag images which have been built in this pipeline from cd project into target project
+        retagImages(context.targetProject, getBuiltImages())
+
+        logger.info "Rolling out ${context.componentId} with HELM, selector: ${options.selector}"
+        refreshResources = true
+        helmUpgrade(context.targetProject)
+
+        if (refreshResources) {
+            deploymentResources = openShift.getResourcesForComponent(
+                context.targetProject, DEPLOYMENT_KINDS, options.selector
+            )
+        }
```

Comment:
The way the variable refreshResources is used make the if condition in line 105 always be true, it does not depends on configuration or previous execution results